nucliadb_core/src/query_language/mod.rs

```diff
@@ -186,41 +186,78 @@ fn split_mixed(expression: BooleanExpression, context: &QueryContext) -> NodeRes
 
 #[derive(Default)]
 pub struct QueryAnalysis {
-    pub prefilter_query: Option<BooleanExpression>,
+    pub labels_prefilter_query: Option<BooleanExpression>,
+    pub keywords_prefilter_query: Option<BooleanExpression>,
     pub search_query: Option<BooleanExpression>,
 }
 
-pub fn translate(query: &str, context: &QueryContext) -> NodeResult<QueryAnalysis> {
-    let Json::Object(subexpressions) = serde_json::from_str(query)? else {
-        #[rustfmt::skip] return Err(node_error!(
-            "Unexpected query format {query}"
-        ));
-    };
-
-    let mut prefilter_queries = Vec::new();
+pub fn translate(
+    labels_query: Option<&str>,
+    keywords_query: Option<&str>,
+    context: &QueryContext,
+) -> NodeResult<QueryAnalysis> {
+    let mut labels_prefilter_queries = Vec::new();
+    let mut keywords_prefilter_queries = Vec::new();
     let mut search_queries = Vec::new();
 
-    for (root_id, expression) in subexpressions {
-        let as_json_expression = serde_json::json!({ root_id: expression });
-        let translation_report = translate_expression(as_json_expression, context)?;
-        if translation_report.has_field_labels && translation_report.has_paragraph_labels {
-            let splitted = split_mixed(translation_report.expression, context)?;
-            prefilter_queries.push(splitted.fields_only);
-            search_queries.push(splitted.paragraphs_only);
-        } else if translation_report.has_field_labels {
-            prefilter_queries.push(translation_report.expression);
-        } else if translation_report.has_paragraph_labels {
-            search_queries.push(translation_report.expression);
+    // Translate labels query. Some of the labels are paragraph-type labels that need to be passed a search query.
+    if let Some(labels_query) = labels_query {
+        if !labels_query.is_empty() {
+            eprintln!("Labels query parsing: {labels_query}");
```

Comment:
Use `tracing` macros instead of `eprintln`

Reply:
sorry, this is a leftover from when I was debugging. I will delete it